# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/WebrtcGmpVideoCodec.cpp
# Commit: bb3495397f18
# Full Hash: bb3495397f18726da8eb3eef7866400c3a495127
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2025-01-24 21:02:20
# Regressor Bug: 1942245
# File Overlap Count: 2
# Description:
#   Bug 1942245 - Remove DataMutex from mInputImageMap. r=aosmond
#   
#   Its use is single threaded.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D234859
# ==============================================================================

diff -r e673cfb778ea -r bb3495397f18 dom/media/webrtc/libwebrtcglue/WebrtcGmpVideoCodec.cpp
--- a/dom/media/webrtc/libwebrtcglue/WebrtcGmpVideoCodec.cpp	Fri Jan 24 11:43:22 2025 +0000
+++ b/dom/media/webrtc/libwebrtcglue/WebrtcGmpVideoCodec.cpp	Fri Jan 24 11:43:23 2025 +0000
@@ -35,8 +35,7 @@
       mFormatParams(aFormat.parameters),
       mCallbackMutex("WebrtcGmpVideoEncoder encoded callback mutex"),
       mCallback(nullptr),
-      mPCHandle(std::move(aPCHandle)),
-      mInputImageMap("WebrtcGmpVideoEncoder::mInputImageMap") {
+      mPCHandle(std::move(aPCHandle)) {
   mCodecParams.mCodecType = kGMPVideoCodecInvalid;
   mCodecParams.mMode = kGMPCodecModeInvalid;
   mCodecParams.mLogLevel = GetGMPLibraryLogLevel();
@@ -354,13 +353,10 @@
     gmp_frame_types.AppendElement(ft);
   }
 
-  {
-    auto inputImageMap = mInputImageMap.Lock();
-    DebugOnly<bool> inserted = false;
-    std::tie(std::ignore, inserted) = inputImageMap->insert(
-        {frame->Timestamp(), {aInputImage.timestamp_us()}});
-    MOZ_ASSERT(inserted, "Duplicate timestamp");
-  }
+  DebugOnly<bool> inserted = false;
+  std::tie(std::ignore, inserted) =
+      mInputImageMap.insert({frame->Timestamp(), {aInputImage.timestamp_us()}});
+  MOZ_ASSERT(inserted, "Duplicate timestamp");
 
   GMP_LOG_DEBUG("GMP Encode: %" PRIu64, (frame->Timestamp()));
   err = mGMP->Encode(std::move(frame), codecSpecificInfo, gmp_frame_types);
@@ -458,14 +454,12 @@
 void WebrtcGmpVideoEncoder::Encoded(
     GMPVideoEncodedFrame* aEncodedFrame,
     const nsTArray<uint8_t>& aCodecSpecificInfo) {
+  MOZ_ASSERT(mGMPThread->IsOnCurrentThread());
   webrtc::Timestamp capture_time = webrtc::Timestamp::Micros(0);
-  {
-    auto inputImageMap = mInputImageMap.Lock();
-    auto handle = inputImageMap->extract(aEncodedFrame->TimeStamp());
-    MOZ_ASSERT(handle);
-    if (handle) {
-      capture_time = webrtc::Timestamp::Micros(handle.mapped().timestamp_us);
-    }
+  auto handle = mInputImageMap.extract(aEncodedFrame->TimeStamp());
+  MOZ_ASSERT(handle);
+  if (handle) {
+    capture_time = webrtc::Timestamp::Micros(handle.mapped().timestamp_us);
   }
 
   MutexAutoLock lock(mCallbackMutex);